airbyte-analytics/src/main/java/io/airbyte/analytics/SegmentTrackingClient.java

```diff
@@ -33,25 +33,30 @@
 import java.util.HashMap;
 import java.util.Map;
 import java.util.function.Supplier;
+import org.apache.logging.log4j.util.Strings;
 
 public class SegmentTrackingClient implements TrackingClient {
 
   private static final String SEGMENT_WRITE_KEY = "7UDdp5K55CyiGgsauOr2pNNujGvmhaeu";
   private static final String AIRBYTE_VERSION_KEY = "airbyte_version";
+  private static final String AIRBYTE_ROLE = "airbyte_role";
 
   // Analytics is threadsafe.
   private final Analytics analytics;
   private final Supplier<TrackingIdentity> identitySupplier;
+  private final Supplier<String> roleSupplier;
 
   @VisibleForTesting
-  SegmentTrackingClient(Supplier<TrackingIdentity> identitySupplier, Analytics analytics) {
+  SegmentTrackingClient(final Supplier<TrackingIdentity> identitySupplier,
+                        final Analytics analytics,
+                        final Supplier<String> roleSupplier) {
     this.identitySupplier = identitySupplier;
     this.analytics = analytics;
+    this.roleSupplier = roleSupplier;
   }
 
   public SegmentTrackingClient(Supplier<TrackingIdentity> identitySupplier) {
-    this.analytics = Analytics.builder(SEGMENT_WRITE_KEY).build();
-    this.identitySupplier = identitySupplier;
+    this(identitySupplier, Analytics.builder(SEGMENT_WRITE_KEY).build(), () -> System.getenv(AIRBYTE_ROLE.toUpperCase()));
```

Comment:
Is there a reason you're not putting this `EnvConfigs`?

Reply:
fixed